src/sparseml/modifiers/obcq/pytorch.py

```diff
@@ -209,3 +209,85 @@ def _infer_mask_block_size(self):
             raise ValueError("mask_structure must be defined")
 
         self.prunen_, self.prunem_ = list(map(int, self.mask_structure.split(":")))
+
+    def _infer_layer_sparsity(self, calibration_dataloader, dev):
+        prev_dev = self.model.device
+        self.model.to(dev)
+        acts = _get_activations(self.model, calibration_dataloader)
+        wanda = []
+        names = []
+        num_w_perlayer = 0
+        for n, m in self.model.named_modules():
+            if isinstance(m, torch.nn.Linear) and "lm_head" not in n:
+                print(f"[DEBUG] owl considering = {n} with shape = {m.weight.shape}")
+                if "layers.1." in n:
+                    num_w_perlayer += 1
```

Comment:
don't love this being hardcoded, it will not generalize well to other architectures. For instance MPT-7b uses `blocks` instead of `layers`

Reply:
Could we use `self.compressible_layers` instead here? That might also solve the regex issue

Reply:
Fixed